.github/workflows/dm_upstream_switch.yaml

```diff
@@ -82,3 +74,69 @@ jobs:
         uses: Ilshidur/action-slack@2.1.0
         with:
           args: "upstream-switch job failed, see https://github.com/pingcap/tiflow/actions/runs/{{ GITHUB_RUN_ID }}"
+
+  upstream-database-switch-mariadb:
+    name: upstream-database-switch-mariadb
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Set up Go env
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.21'
+
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Cache go modules
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-ticdc-${{ hashFiles('go.sum') }}
+
+      - name: Cache Tools
+        id: cache-tools
+        uses: actions/cache@v2
+        with:
+          path: tools/bin
+          key: ${{ runner.os }}-ticdc-tools-${{ hashFiles('tools/check/go.sum') }}
+
+      - name: Build DM binary
+        run: make dm_integration_test_build
+
+      - name: Setup containers
+        run: |
+          docker-compose -f ./dm/tests/upstream_switch_mariadb/docker-compose.yml up -d
+
+      - name: Run test cases
+        run: |
+          bash ./dm/tests/upstream_switch_mariadb/case.sh
+
+      - name: Copy logs to hack permission
+        if: ${{ always() }}
+        run: |
+          mkdir ./logs
+          sudo cp -r -L /tmp/dm_test/upstream_switch_mariadb/master/log ./logs/master
+          sudo cp -r -L /tmp/dm_test/upstream_switch_mariadb/worker1/log ./logs/worker1
+          sudo cp -r -L /tmp/dm_test/upstream_switch_mariadb/worker2/log ./logs/worker2
+          sudo chown -R runner ./logs
+
+      # Update logs as artifact seems not stable, so we set `continue-on-error: true` here.
+      - name: Upload logs
+        continue-on-error: true
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: upstream-switch-mariadb-logs
+          path: |
+            ./logs
+
+      # send Slack notify if failed.
+      # NOTE: With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository.
+      - name: Slack notification
+        if: ${{ failure() }}
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_NOTIFY }}
```

Comment:
no one use slack now, could we use feishu instead

Reply:
lgtm

Reply:
I'll delete them first
<img width="338" alt="image" src="https://github.com/pingcap/tiflow/assets/23556834/98f26bf4-ccb9-448d-bf05-a81966f3dec1">